src/sagas/vmChanges.js

```diff
@@ -177,54 +162,132 @@ function* composeProvisionSourceTemplate ({ vm, basic, disks }) {
 
   /*
    * If a template defined disk needs to be created in a storage domain different than
-   * the one defined in the template, or if the disk's sparse value is changed, the
-   * changes need to passed along in the VM create call.
+   * the one defined in the template, or if the disk's format/sparse values need to be
+   * changed, pass those changes along in the VM create call.
    *
    * See: http://ovirt.github.io/ovirt-engine-api-model/master/#services/vms/methods/add
    */
-  disks
+  const storageDomains = yield select(state => state.storageDomains) // { [id]: sd }
+  const disksFromTemplate = disks
     .filter(disk => disk.isFromTemplate)
-    .forEach(disk => {
-      const templateDisk = template.get('disks').find(tdisk => tdisk.get('id') === disk.id)
-      if (!templateDisk) {
-        return
+    .map(disk => ({ disk, baseDisk: template.get('disks').find(tdisk => tdisk.get('id') === disk.id) }))
+
+  const templateOptimizedFor = template.get('type')
+  const vmOptimizedFor = basic.optimizedFor
+  const storageAllocation =
```

Comment:
See this comment https://github.com/oVirt/ovirt-web-ui/pull/1508/files#r701252466 for some more info about storage allocation.  I can probably drop it from the sagas since it is really doesn't impact any other selections.  I needed it to be there so I could match up logic with webadmin's behavior properly.